server/schema.ts

```diff
@@ -1,40 +1,78 @@
-const Fuse = require('fuse.js');
+import { Fuse } from 'fuse.js';
 
-const {
+import {
   GraphQLInt,
   GraphQLList,
   GraphQLNonNull,
   GraphQLObjectType,
+  GraphQLObjectTypeConfig,
   GraphQLSchema,
   GraphQLString,
-} = require('graphql');
-const {
+} from 'graphql';
+import {
+  Author,
   authorType,
   cardType,
+  Category,
   categoryType,
-} = require('./types/');
+} from './types/';
+
+export interface AuthorJSON {
+  [authorId: string]: Author;
+}
+
+export interface CategoryJSON {
+  [categoryId: string]: Category;
+}
+
+export interface SingleCardJSON {
+  authors: Array<keyof AuthorJSON>;
+  categories: Array<keyof CategoryJSON>;
+  content: string;
+  created: number;
+  drugs: string[]|null;
+  id: string;
+  title: string;
+  updates: number[]|null;
+}
+
+export interface CardJSON {
+  [cardId: string]: SingleCardJSON;
+};
+
+export interface Context {
+  entities: {
+    authors: AuthorJSON;
+    categories: CategoryJSON;
+    cards: CardJSON;
+  };
+  result: Array<keyof CardJSON>;
+};
+
+export type RootContext = GraphQLObjectTypeConfig<any, Context>;
 
 /**
- * queryType has the following shape:
- *   type Query {
- *     authors(): [Author]
- *     author(id: String!): Author
- *     cards(category: String, drug: String): [Card]
- *     card(id: String!): Card
- *     categories(): [Category]
- *     recentlyAdded(n: Int): [Card]
- *     recentlyUpdated(n: Int): [Card]
- *     neverUpdated(n: Int): [Card]
- *     oldestUpdates(n: Int): [Card]
- *     search(input: String): [Card]
- *   }
+ * Query Type Definition
+ *
+ *    type Query {
+ *      authors(): [Author]
+ *      author(id: String!): Author
+ *      cards(category: String, drug: String): [Card]
+ *      card(id: String!): Card
+ *      categories(): [Category]
+ *      recentlyAdded(n: Int): [Card]
+ *      recentlyUpdated(n: Int): [Card]
+ *      neverUpdated(n: Int): [Card]
+ *      oldestUpdates(n: Int): [Card]
+ *      search(input: String): [Card]
+ *    }
  */
-const queryType = new GraphQLObjectType({
+const queryType = new GraphQLObjectType(<RootContext>{
   name: 'Query',
   fields: () => ({
     authors: {
       type: new GraphQLList(authorType),
-      resolve: (root, args, context) => Object.values(context.entities.authors),
+      resolve: (_root, _args, context) => Object.values(context.entities.authors),
```

Comment:
Because we have `noUnusedParam` enabled in `tsconfig.json`, you have to explicitly tell the compiler that you are aware that the param in question is not going to be used (typically this only comes up when dealing with external libraries). The way to tell the compiler that you are aware of the unused param is be prefixing `_`.